Algorithm.CSharp/TrainExampleAlgorithm.cs

```diff
@@ -0,0 +1,120 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using System;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This example shows how we can execute a method that takes longer than Lean timeout limit
+    /// This feature is useful for algorithms that train models
+    /// </summary>
+    /// <meta name="tag" content="using quantconnect" />
+    public class TrainExampleAlgorithm : QCAlgorithm
```

Comment:
Nice example algorithm! I think it would be nice to add an algorithm using this feature into the regression algorithms, `IRegressionAlgorithmDefinition`, so we test it continuously